lib/spack/spack/util/executable.py

```diff
@@ -94,6 +94,9 @@ def __call__(self, *args, **kwargs):
 
         Keyword Arguments:
             env (dict): The environment to run the executable with
+            env_dump (dict): Dict to be set to the environment actually used
```

Comment:
A couple requests:

* I think this should be `dump_env` (there are no references to `env_dump`, but there are to `dump_env`)
* This appears to be used only for testing. IMO it's worth clarifying that in the explanation and also adding a `_` to the prefix, i.e. `_dump_env`.

Reply:
@scheibelp I wasn't sure what the policy on testing-only was, but I can do that.